src/app/features/container/container.tsx

```diff
@@ -69,9 +69,11 @@ export function Container() {
   const displayHeader = !isLandingPage(pathname) && !isNoHeaderPopup(pathname);
   const isSessionLocked = getIsSessionLocked(pathname);
 
+  // TODO: Refactor? This is very hard to manage with dynamic routes. Temporarily
```

Comment:
I agree, this isn't great. Before we were storing the whole header including callback in the state. 

Maybe we could try using the location state to store where it should go back to? 

For some of the other dynamic routes I was using `match` to get to more specific cases. 

If we use `includes` here it also means if you reach the `/swap/review` page then you go all the way back to the homepage on back and not to the previous page where you are selecting what to swap. 

In those cases I think it should go back `navigate(../)`

Reply:
Rather than trying to put all the go back logic in the container—which think we all agree isn't a scalable solution—can't we instead find a way to define this in each component tree where there is a back button?

What if the header back button simply publishes a "go back" event, e.g. with PubSub pattern, and components, via a hook listen for this event? 

```ts
function useHandleGoBack (handler) {
  // listen for go back logic, then fire handler()
}
export function Swap() {
  // nav passed as handler to avoid useNavigate if not needed
  useHandleGoBack(navigate => navigate(RouteUrls.Home))
```

Perhaps not perfect. We'd need to be careful not to have double handlers, and ensure we define goback where necessary, but this demonstrates a way we can completely decouple the logic.

Any other ideas?

Reply:
Opened issue for this here

Reply:
Can you make an issue to refactor this based on your understanding for what you wrote here and your implementation? I will likely not tackle this here.

Reply:
Nvm, my comments hadn't refreshed, sorry that last comment was outdated.